packages/scene-composer/src/utils/entityModelUtils/motionIndicatorComponent.spec.ts

```diff
@@ -0,0 +1,370 @@
+      ...indicatorBase,
+      config: { ...indicatorBase.config, defaultSpeed: 11 },
+    });
+
+    expect(result.properties!['config_defaultSpeed']).toEqual({
+      value: { integerValue: 11 },
+    });
+  });
+
+  it('should return expected motion indicator component with default background color', () => {
+    const result = createMotionIndicatorEntityComponent({
+      ...indicatorBase,
+      config: { ...indicatorBase.config, defaultBackgroundColor: '#123' },
+    });
+
+    expect(result.properties!['config_defaultBackgroundColor']).toEqual({
+      value: { stringValue: '#123' },
+    });
+  });
+
+  it('should return expected motion indicator component with default foreground color', () => {
+    const result = createMotionIndicatorEntityComponent({
+      ...indicatorBase,
+      config: { ...indicatorBase.config, defaultForegroundColor: '#abc' },
+    });
+
+    expect(result.properties!['config_defaultForegroundColor']).toEqual({
+      value: { stringValue: '#abc' },
+    });
+  });
+
+  it('should return expected tag component with speed binding', () => {
+    const result = createMotionIndicatorEntityComponent({
+      ...indicatorBase,
+      valueDataBindings: {
+        speed: { ruleBasedMapId: 'rule-id', valueDataBinding: { dataBindingContext: { entityId: 'eid' } } },
+      },
+    });
+
+    expect(result.properties!['appearanceBindings']).toEqual({
+      value: {
+        listValue: [
+          {
+            mapValue: {
+              ruleBasedMapId: {
+                stringValue: 'rule-id',
+              },
+              bindingName: {
+                stringValue: 'speed',
+              },
+              entityId: {
+                stringValue: 'eid',
+              },
+            },
+          },
+        ],
+      },
+    });
+  });
+
+  it('should return expected tag component with background color binding', () => {
+    const result = createMotionIndicatorEntityComponent({
+      ...indicatorBase,
+      valueDataBindings: {
+        backgroundColor: { valueDataBinding: { dataBindingContext: { entityId: 'eid' } } },
+      },
+    });
+
+    expect(result.properties!['appearanceBindings']).toEqual({
+      value: {
+        listValue: [
+          {
+            mapValue: {
+              bindingName: {
+                stringValue: 'backgroundColor',
+              },
+              entityId: {
+                stringValue: 'eid',
+              },
+            },
+          },
+        ],
+      },
+    });
+  });
+
+  it('should return expected tag component with multiple bindings', () => {
+    const result = createMotionIndicatorEntityComponent({
+      ...indicatorBase,
+      valueDataBindings: {
+        foregroundColor: { valueDataBinding: { dataBindingContext: { entityId: 'eid' } } },
+        speed: { ruleBasedMapId: 'rule-id', valueDataBinding: { dataBindingContext: { entityId: 'eid' } } },
+      },
+    });
+
+    expect(result.properties!['appearanceBindings']).toEqual({
```

Comment:
Nit-pick: It would likely improve readability if this object was predefined as a `const` with a clear name that let's the reader know generally what it is. Very much a preference statement & non-blocking.

Reply:
I'll consider that next time